### PR TITLE
Update T slot fasteners

### PR DESCRIPTION
**Pull request by hasecilu**
## Update T slot nuts to change the slot width.
Previously the slot width of the T slot nuts was fixed in the data table. In the case of the [GN507 nut](https://www.jwwinco.com/en-us/products/3.2-Mounting-positioning-levelling-with-screws-clamping-and-supporting-elements/Nuts-Bolts-for-T-Slots/GN-507-Steel-T-Nuts-For-Aluminum-Extrusions) the manufacturer allows different slot widhts with different thread sizes.

Now is possible to change the slot width and the thread size independently.
- Script updated
- Data tables were updated
  - the GN507 data table carries the logic to this new feature and
  - ~the DIN508 has a dummy table because the DIN508 norm assigns one slot width to one thread~
- Some variables were renamed

~Just there is a problem: I added the `AddTSlotNut(self, obj, cnt)` method to have custom output on the BOM but something is missing and I don't know what is.~ **Solved**

## New fasteners
Main data is [from here](https://www.jwwinco.com/en-us/products/3.2-Mounting-positioning-levelling-with-screws-clamping-and-supporting-elements/Nuts-Bolts-for-T-Slots) (data for 3030 and 4040 series) but there is extra data that can be found with retailers, specifically data about 2020-series Al extrusion compatible fasteners
- GN505
- GN505.4

image

### CODE REVIEW

**Comment by shaise** [comment]
Very nice work!!!
Please let me know when it is ready to be merged. Also, please make sure that old files with T-Slots can be loaded, and parameters can be changed without breaking.

**Comment by hasecilu** [comment]
I have made a test with old files and found that the project don't break, but
- The list of available diameters is not updated, i.e. the new ones are missing, for example: no M27 on DIN 508 and no M3 on GN 507
- The previous slot W parameter appears but does nothing (as expected)
- The new slot Width parameter is working with the available diameters
- The output in the BOM is the old one when marked only as nut
- In the type parameter the new GN 505 nut don't appear

As long as you don't want a new added diameter with a previously created nut it seems to be not problem.

Apart from that it should be ready to merge.

File: FastenersCmd.py [hunk 1]
```diff
@@ -715,5 +728,6 @@ def onDocumentRestored(self, obj):
 FastenerBase.FSAddFastenerType("Stud")
 FastenerBase.FSAddFastenerType("HeatSet", False)
 FastenerBase.FSAddFastenerType("RetainingRing", False)
+FastenerBase.FSAddFastenerType("T-Slot", False)
```
Comment (hasecilu) [inline FastenersCmd.py:731]: Is really needed this line, what changes?
I think it should be "TSlot" instead of "T-Slot".

Comment (shaise) [reply FastenersCmd.py:731]: the types are used for example in fschangeparams.py:
image

**Comment by shaise** [comment]
Meged! Thanks a lot!

**Pull request merged**